From 1be2bafe7a9aabf0389da9e685051859c56e64ce Mon Sep 17 00:00:00 2001
From: Kate Hsuan <hpa@redhat.com>
Date: Thu, 9 Feb 2023 15:26:37 +0800
Subject: [PATCH] Set a new ID offset for the private event

According to the v4l2loopback, the private event ID should be
start from V4L2_EVENT_PRIVATE_START+0x08E00000. Therefore, the
V4L2_EVENT_PRI_CLIENT_USAGE is V4L2_EVENT_PRIVATE_START+
0x08E00000+1

Signed-off-by: Kate Hsuan <hpa@redhat.com>
---
 src/v4l2-relayd.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/v4l2-relayd.c b/src/v4l2-relayd.c
index 21bb0d5..d731af0 100644
--- a/src/v4l2-relayd.c
+++ b/src/v4l2-relayd.c
@@ -27,7 +27,10 @@
 #include <gst/app/gstappsrc.h>
 #include <gst/video/video-info.h>
 
-#define V4L2_EVENT_PRI_CLIENT_USAGE  V4L2_EVENT_PRIVATE_START
+#define V4L2LOOPBACK_EVENT_OFFSET 0x08E00000
+#define V4L2_EVENT_PRI_CLIENT_USAGE  (V4L2_EVENT_PRIVATE_START+\
+                                      V4L2LOOPBACK_EVENT_OFFSET+\
+                                      1)
 
 struct v4l2_event_client_usage {
   __u32 count;
-- 
2.39.2

